spec/environment_spec.rb

```diff
@@ -545,6 +550,131 @@ def promote_content(content1, env1)
+    product = create_product
+    content1 = new_content(1)
+    content2 = new_content(2)
+
+    @cp.add_content_to_product(@owner['key'], product['id'], content1['id'])
+    @cp.add_content_to_product(@owner['key'], product['id'], content2['id'])
+
+    env1 = @cp.create_environment(@owner['key'], random_string("env1"), random_string("test_env_1"))
+    env2 = @cp.create_environment(@owner['key'], random_string("env2"), random_string("test_env_2"))
+    promote_content(content1, env1)
+    promote_content(content2, env2)
+    consumer = @cp.register(random_string("consumer2"), :system, nil,
+                            facts={'system.certificate_version' => '3.3', "uname.machine" => "x86_64"}, random_string("consumer2"),
+                            @owner['key'], [], [], [{ 'id' => env1['id']}, { 'id' => env2['id']}])
+    consumer_cp = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+
+    @cp.delete_environment(env1['id'])
+
+    pool = @cp.create_pool(@owner['key'], product['id'])
+    consumer_cp.consume_pool(pool['id'], {:quantity => 1})[0]
+    certs = consumer_cp.list_certificates
+    certs.length.should == 1
+
+    json_body = extract_payload(certs[0]['cert'])
+    json_body['products'].length.should == 1
+    json_body['products'][0]['content'].length.should == 1
+  end
+
+  it 'should revoke entitlements granted by the deleted environment' do
+    product1 = create_product
+    product2 = create_product
+    content1 = new_content(1)
+    content2 = new_content(2)
+
+    @cp.add_content_to_product(@owner['key'], product1['id'], content1['id'])
+    @cp.add_content_to_product(@owner['key'], product2['id'], content2['id'])
+
+    env1 = @cp.create_environment(@owner['key'], random_string("env1"), random_string("test_env_1"))
+    env2 = @cp.create_environment(@owner['key'], random_string("env2"), random_string("test_env_2"))
+    promote_content(content1, env1)
+    promote_content(content2, env2)
+    consumer = @cp.register(random_string("consumer2"), :system, nil,
+                            facts={'system.certificate_version' => '3.3', "uname.machine" => "x86_64"}, random_string("consumer2"),
+                            @owner['key'], [], [], [{ 'id' => env1['id']}, { 'id' => env2['id']}])
+    consumer_cp = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+
+    pool1 = @cp.create_pool(@owner['key'], product1['id'])
+    pool2 = @cp.create_pool(@owner['key'], product2['id'])
+    consumer_cp.consume_pool(pool1['id'], {:quantity => 1})[0]
+    consumer_cp.consume_pool(pool2['id'], {:quantity => 1})[0]
+    certs = consumer_cp.list_certificates
+    certs.length.should == 2
+
+    @cp.delete_environment(env1['id'])
+
+    certs = consumer_cp.list_certificates
+    certs.length.should == 2
+
+    contents = contents_of(certs[0]) + contents_of(certs[1])
+    contents.size.should == 1
+  end
+
+  it 'should revoke entitlements granted by the demoted content' do
```

Comment:
This is a duplicate of `'regenerates certificates after promoting/demoting content'` in `environment_cert_v3.rb`

Reply:
Done.